llvm/lib/Analysis/TargetLibraryInfo.cpp

```diff
@@ -300,6 +300,7 @@ static void initializeLibCalls(TargetLibraryInfoImpl &TLI, const Triple &T,
       TLI.setUnavailable(LibFunc_expf);
       TLI.setUnavailable(LibFunc_floorf);
       TLI.setUnavailable(LibFunc_fmodf);
+      TLI.setUnavailable(LibFunc_hypotf);
```

Comment:
I also have the same question because I also confused for that. I finally chose that place because I followed https://github.com/llvm/llvm-project/pull/99611 (I'm sorry. It's my first time contributing to that. So I chose to follow other people's way.) 

According to https://en.cppreference.com/w/c/numeric/math/remquo, the remquo series functions seems introduced in C99 but this commit adds remquo to the C89 code block I added the hypot to.
https://github.com/llvm/llvm-project/blob/6effab990c5c1b4fe55fcd43004a1fd88145bb8d/llvm/lib/Analysis/TargetLibraryInfo.cpp#L291-L319

Reply:
These functions are treated as a part of the system, not the language

Reply:
According to page 191 of https://web.archive.org/web/20200909074736if_/https://www.pdf-archive.com/2014/10/02/ansi-iso-9899-1990-1/ansi-iso-9899-1990-1.pdf, it seems the C89 doesn't have the C math functions with ’f‘ and 'l' suffixes.

Reply:
> These functions are treated as a part of the system, not the language

I see. It solves my confusion.